credmon/utils/utils.py

```diff
@@ -256,26 +256,39 @@ def generate_secret_key():
 
     keyfile = os.path.join(htcondor.param.get("SEC_CREDENTIAL_DIRECTORY", "/var/lib/condor/credentials"), "wsgi_session_key")
 
+    # Create the secret key file, if possible, with read-only permissions, if it doesn't exist
     try:
-        fd = os.open(keyfile, os.O_CREAT | os.O_RDWR, 0o600)
-        current_key = os.read(fd, 24)
+        os.close(os.open(keyfile, os.O_CREAT | os.O_EXCL | os.O_RDWR, stat.S_IRUSR))
     except Exception:
-        logger.warning("Unable to access WSGI session key; will use a non-persistent key")
+        # An exception will be thrown if it exists, and that's fine, we'll check that we can read it at the next step.
```

Comment:
Catch the precise exception, not `Exception`, if you are expecting a specific one.

Reply:
Resolved in https://github.com/htcondor/scitokens-credmon/pull/21/commits/7aad890d92a1716ce522e9524b91c6dc8d9bc25c